src/ui/components/dashboard/ApplicationModal.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {
+  Icon,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+} from '@chakra-ui/react';
+import PropTypes, {bool, func, string} from 'prop-types';
+import PrimaryButton from '../common/buttons/PrimaryButton';
+import SecondaryButton from '../common/buttons/SecondaryButton';
+import {MdOutlineSave} from 'react-icons/md';
+import {Form, Formik} from 'formik';
+import * as Yup from 'yup';
+import {APPLICATION_STATUS_MAP} from './StatusLabel';
+import TextInput from '../common/forms/TextInput';
+import TextAreaInput from '../common/forms/TextAreaInput';
+import DateInput from '../common/forms/DateInput';
+import SelectInput from '../common/forms/SelectInput';
+
+// These will be deleted once we can pull the data from the database
+const companyOptions = [
+  <option value="[UUID1]" key="Adobe">
+    Adobe
+  </option>,
+  <option value="[UUID2]" key="Capital One">
+    Capital One
+  </option>,
+  <option value="[UUID3]" key="Chase Bank">
+    Chase Bank
+  </option>,
+  <option value="[UUID4]" key="Chipotle">
+    Chipotle
+  </option>,
+];
+
+const VALID_STATUSES = Object.keys(APPLICATION_STATUS_MAP).filter(key => key !== 'error');
+
+export const statusOptions = VALID_STATUSES.map(key => {
+  return (
+    <option value={key} key={key}>
+      {APPLICATION_STATUS_MAP[key].text}
+    </option>
+  );
+});
+
+// returns users date in format 'YYYY-MM-DD'
+const getTodaysDate = () => {
+  const timezoneOffsetMinutes = new Date().getTimezoneOffset();
+  const userDate = new Date(new Date().getTime() - timezoneOffsetMinutes * 60 * 1000);
+  return userDate.toISOString().split('T')[0];
+};
+
+// formats a string of skills into an array of strings
+// IE: "skill 1, skill 2, skill 3" -> ["skill 1", "skill 2", "skill 3"]
+const formatSkills = skillsString => {
+  if (skillsString.length === 0) return [];
+  const skills = skillsString.split(',');
+  return skills.map(skill => skill.trim());
```

Comment:
Could you also just add something here before you return that filters out empty strings?

I was testing this and found that ["javascript, python,"] will become ["javascript", "python", ""]

Reply:
Good catch!